### PR TITLE
More tests about exceptions

### PR DESCRIPTION
**Pull request by antocuni**
Add two tests which were failing on PyPy, testing that the functions return the correct error_value in case of error

### CODE REVIEW

File: test/test_hpyerr.py [hunk 1]
```diff
@@ -341,3 +341,49 @@ def check_warning(cls):
...
+                res = HPy_TrueDivide(ctx, a, b);
+
+                // the point of the test is to check that in case of error
+                // HPy_Div returns HPy_NULL
+                if (HPy_IsNull(res)) {
+                    HPyErr_Clear(ctx);
+                    return HPyLong_FromLong(ctx, -42);
+                }
+
+                HPy_Close(ctx, b);
```
Comment (hodgestar) [inline test/test_hpyerr.py:364]: `b` should not be closed here.

**Comment by hodgestar** [comment]
Otherwise looks good!

File: test/test_hpyerr.py (hunk 1 above)
Comment (antocuni) [reply test/test_hpyerr.py]: uh, true! We need the debug mode :facepalm:

Comment (antocuni) [reply test/test_hpyerr.py]: fixed

**Review by hodgestar — APPROVED** [review]

**Pull request merged**